iterations_runner.lua

```diff
@@ -121,8 +121,9 @@ if #arg ~= 4 and #arg ~= 7 then
 end
 
 -- Add the benchmarks directory as a Lua module search path
-local benchdir = string.match(BM_benchmark, "^(.-)[/]?([^/]*)$")
-if benchdir then
+local benchdir, scriptname = string.match(BM_benchmark, "^(.-)[/]?([^/]*)$")
+if benchdir and scriptname then
+    assert(benchdir:sub(#benchdir, -1) ~= "/")
```

Comment:
Is there a Lua function that says "is this path $p$ a dir". e.g. in Python it would be `os.isdir(p)`. What we don't want to do is add something to the search path that isn't a dir: it'll look like it works, but silently fail in a confusing way.

Reply:
There is pretty much no built-in file system API only basic file IO

Reply:
There's no way using the standard library to do this?

Reply:
well you can check that a file exists by trying to open it in Lua but i don't know anyway to check a that directory exists with the built-in api.

Reply:
Can you have a look around to see if this is possible? If it absolutely isn't, then there isn't much we can do about it (even though I'm a bit stunned...).

Reply:
The only ways seem to be rename the dir to its self with os.rename(dirname, dirname) and check for an error or execute a shell command like 'cd' and check the result

Reply:
Renaming a directory? Erp. No, we definitely don't want to do that! OK, so Lua's standard library sucks in this regard :( Oh well. Let's remove the assert -- it's not useful as it stands. We'd better just cross our fingers that we don't encounter problems...